core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -329,6 +337,42 @@ public synchronized TopLevelItem createProject( TopLevelItemDescriptor type, Str
     }
 
     /**
+     * Populate a {$link Categories} from a specific {$link ItemGroup}.
+     *
+     * @return
+     */
+    public static Categories getCategories(Authentication a, ItemGroup c) {
```

Comment:
:bug: from what I see there is a potential synchronization issue on the Categories:items container.